packages/use-contractkit/src/utils/metamask.ts

```diff
@@ -0,0 +1,233 @@
+import { ContractKit, newKit } from '@celo/contractkit';
+import { GoldTokenWrapper } from '@celo/contractkit/lib/wrappers/GoldTokenWrapper';
+import { StableTokenWrapper } from '@celo/contractkit/lib/wrappers/StableTokenWrapper';
+import Web3 from 'web3';
+
+import { Alfajores, Baklava, Mainnet } from '../constants';
+import { Ethereum } from '../global';
+import { ChainId, Network } from '../types';
+
+const CELO_PARAMS = Object.freeze({
+  chainName: 'Celo',
+  testnet: false,
+  nativeCurrency: {
+    name: 'CELO',
+  },
+});
+
+const ALFAJORES_PARAMS = Object.freeze({
+  chainName: 'Alfajores Testnet',
+  testnet: true,
+  nativeCurrency: {
+    name: 'A-CELO',
+  },
+});
+
+const BAKLAVA_PARAMS = Object.freeze({
+  chainName: 'Baklava Testnet',
+  testnet: true,
+  nativeCurrency: {
+    name: 'B-CELO',
+  },
+});
+
+const params: { [chain in ChainId]: typeof CELO_PARAMS } = {
+  [ChainId.Mainnet]: CELO_PARAMS,
+  [ChainId.Alfajores]: ALFAJORES_PARAMS,
+  [ChainId.Baklava]: BAKLAVA_PARAMS,
+};
+
+interface ERC20Token {
+  address: string;
+  name: string;
+  symbol: string;
+  decimals: number;
+  image?: string;
+}
+
+interface AddERC20TokenParameter {
+  type: 'ERC20';
+  options: ERC20Token;
+}
+
+interface CeloTokens {
+  CELO: GoldTokenWrapper;
+  cUSD: StableTokenWrapper;
+  cEUR: StableTokenWrapper;
+}
+type StableTokens = Omit<CeloTokens, 'CELO'>;
+
+interface AddEthereumChainParameter {
+  chainId: string;
+  chainName: string;
+  nativeCurrency: Omit<ERC20Token, 'address'>;
+  rpcUrls: string[];
+  blockExplorerUrls?: string[];
+  iconUrls?: string[]; // Currently ignored by metamask
+}
+
+const NETWORKS = {
+  [ChainId.Mainnet]: Mainnet,
+  [ChainId.Alfajores]: Alfajores,
+  [ChainId.Baklava]: Baklava,
+};
+
+enum MetamaskRPCErrorCode {
+  AwaitingUserConfirmation = -32002,
+  UnknownNetwork = 4902,
+}
+
+interface MetamaskRPCError {
+  code: MetamaskRPCErrorCode;
+  message: string;
+  stack: string;
+}
+
+const makeNetworkParams = async (
+  info: Network,
+  CELO: GoldTokenWrapper
+): Promise<AddEthereumChainParameter> => {
+  const [symbol, decimals, name] = await Promise.all([
+    CELO.symbol(),
+    CELO.decimals(),
+    CELO.name(),
+  ]);
+
+  return {
+    chainId: `0x${info.chainId.toString(16)}`,
+    chainName: params[info.chainId].chainName ?? info.name,
+    nativeCurrency: {
+      name: params[info.chainId].nativeCurrency.name,
+      symbol,
+      decimals,
+    },
+    rpcUrls: [info.rpcUrl],
+    blockExplorerUrls: [info.explorer],
+    iconUrls: [`https://celoreserve.org/assets/tokens/${symbol}.svg`], // unfortunately unused
```

Comment:
this means that in  the metamask frame, it'll look like this rather than use the nice icon
![Screenshot 2021-12-09 at 17 58 22](https://user-images.githubusercontent.com/3650621/145441519-ef2f5ca7-31b8-4ca8-b481-2df88da9417f.png)